src/lib.rs

```diff
@@ -1791,40 +1801,31 @@ impl Connection {
             {
                 let frame = frame::Frame::HandshakeDone;
 
-                payload_len += frame.wire_len();
-                left -= frame.wire_len();
-
-                frames.push(frame);
-
-                self.handshake_done_sent = true;
+                if add_frame! {frames, frame, payload_len, left} {
+                    self.handshake_done_sent = true;
+                }
             }
 
             // Create MAX_STREAMS_BIDI frame.
             if self.streams.should_update_max_streams_bidi() {
                 let max = self.streams.update_max_streams_bidi();
                 let frame = frame::Frame::MaxStreamsBidi { max };
 
-                payload_len += frame.wire_len();
-                left -= frame.wire_len();
-
-                frames.push(frame);
-
-                ack_eliciting = true;
-                in_flight = true;
+                if add_frame! {frames, frame, payload_len, left} {
```

Comment:
`update_max_streams_bidi()` is not idempotent, and if this fails we'll not try sending a MAX_STREAMS frame again. `update_max_streams_bidi()` needs to be split into two, with one function returning the new value, and the other updating the internal value.

Reply:
Good point. I split into `update_max_streams_bidi/uni()` to return the next value and `max_streams_bidi/uni_next()` to get next value for creating a frame.